packages/core/src/standards/http.ts

```diff
@@ -796,7 +796,11 @@ export async function fetch(
       headers,
     });
   } else {
-    res = new Response(baseRes.body, baseRes);
+    // https://fetch.spec.whatwg.org/#null-body-status
+    res = new Response(
```

Comment:
I am very new to miniflare, but there are so many things happening in this function.
This is already an improvement in readability IMO:

```js
const body = [101, 204, 205, 304].includes(baseRes.status) ? null : baseRes.body;
res = new Response(body, baseRes);
```

but the whole if/else could be moved to a separate function that would simply return the new response


```js
function toHybridResponse(baseRes: Response): Response { // or any other naming that fits the project conventions
  if (baseRes.type !== "opaqueredirect") {
    const body = [101, 204, 205, 304].includes(baseRes.status) ? null : baseRes.body;
    return new Response(body, baseRes);
  }

  // Unpack opaque responses. This restriction isn't needed server-side,
  // and Cloudflare doesn't support Response types anyway.
  // @ts-expect-error symbol properties are not included in type definitions
  const internalResponse = baseRes[fetchSymbols.kState].internalResponse;
  const headersList = internalResponse.headersList;
  assert(headersList.length % 2 === 0);
  const headers = new Headers();
  for (let i = 0; i < headersList.length; i += 2) {
    headers.append(headersList[i], headersList[i + 1]);
  }
  // Cloudflare returns a body here, but undici aborts the stream so
  // unfortunately it's unusable :(
  return new Response(null, {
    status: internalResponse.status,
    statusText: internalResponse.statusText,
    headers,
  });
}
```